### PR TITLE
fix(twig): improve type of `extendFilter` params

### PR DESCRIPTION
**Pull request by G-Rath**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/twigjs/twig.js/wiki/Extending-twig.js#filters
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

Have tested this locally - Twig will pass an array if there are brackets, otherwise it'll pass `false`.

### CODE REVIEW

File: types/twig/index.d.ts [hunk 1]
```diff
@@ -54,7 +54,7 @@ export interface RenderOptions {
 }
 
 export function twig(params: Parameters): Template;
-export function extendFilter(name: string, definition: (left: any, ...params: any[]) => string): void;
+export function extendFilter(name: string, definition: (left: any, params: any[] | false) => string): void;
```
Comment (emmanuelgautier) [inline types/twig/index.d.ts:57]: Following the wiki, the `params` argument is optional (https://github.com/twigjs/twig.js/wiki/Extending-twig.js#filters).

You should only put the argument optional.

File: types/twig/twig-tests.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ const compOpts: twig.CompileOptions = {
     }
 };
 
-twig.extendFilter(str, (left: any, ...params: any[]) => {
+twig.extendFilter(str, (left: any, params: any[] | false) => {
```
Comment (emmanuelgautier) [inline types/twig/twig-tests.ts:41]: According to the first comment, you should add here a new line trying to call the function but with the `params` argument

**Review by emmanuelgautier — CHANGES_REQUESTED** [review]

File: types/twig/index.d.ts (hunk 1 above)
Comment (G-Rath) [reply types/twig/index.d.ts:57]: @emmanuelgautier the documentation is showing a function that doesn't define the last argument, which is not the same as the argument being optional; in JS it's valid for functions to not define the last argument(s) if they're being used.

In my testing, Twig _always_ passes a value for `params`, so it shouldn't be marked as optional here.

File: types/twig/twig-tests.ts (hunk 2 above)
Comment (G-Rath) [reply types/twig/twig-tests.ts:41]: As I said in my reply to the first comment, that's not true - but I will add a test matching the one in the docs to ensure omitting the second argument doesn't cause an error :)

**Review by emmanuelgautier — APPROVED** [review]

**Comment by G-Rath** [comment]
Ready to merge

**Pull request merged**